### PR TITLE
Setting init delay functions

### PR DESCRIPTION
**Pull request by PabstMirror**
From https://github.com/acemod/ACE3/issues/1842#issuecomment-120689590

Adds a way to run a function w/ parameters after settings are initialized.

We really need something like this for when we run code on objects that depends on settings, as the order isn't guaranteed in dedicated MP.

For example: @ViperMaul @Glowbal - the new medical FUNC(init) can run before settings are init
https://github.com/acemod/ACE3/blob/medical-adjustments/addons/medical/functions/fnc_init.sqf

@jonpas did something very similar in https://github.com/acemod/ACE3/pull/2187/files#diff-75f4a28dd014c00256df94dd2f6e929fR28

Helper function, otherwise you can just do something like:
```
if (!EGVAR(common,settingsInitFinished)) exitWith {
    EGVAR(common,runAtSettingsInitialized) pushBack [FUNC(init), _this];
};
```

### CODE REVIEW

**Comment by jonpas** [comment]
:+1:

File: addons/common/XEH_postInit.sqf [hunk 1]
```diff
@@ -150,6 +150,15 @@ call FUNC(checkFiles);
     //Event that settings are safe to use:
     ["SettingsInitialized", []] call FUNC(localEvent);
 
+    //Set init finished and run all delayed functions:
+    GVAR(settingsInitFinished) = true;
+    diag_log text format ["[ACE] %1 delayed functions running", (count GVAR(runAtSettingsInitialized))];
```
Comment (Glowbal) [inline addons/common/XEH_postInit.sqf]: Should probably make use of the new logging macro(s) instead.

**Comment by SilentSpike** [comment]
I like it, but I think we should condense the function name

**Comment by SilentSpike** [comment]
All public functions that rely on settings should probably make use of this, since people can be running stuff from init fields.

**Comment by commy2** [comment]
InitPostPost

**Comment by SilentSpike** [comment]
PostSettings

**Comment by Glowbal** [comment]
Need this to fix some issues with medical initalization.

**Pull request merged**